### PR TITLE
Add documentation for customizing the webhook

### PR DESCRIPTION
**Pull request by JonCrowther**
Taking over from https://github.com/rancher/rancher-docs/pull/1081

Fixes https://github.com/rancher/rancher/issues/43621

## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, make sure to target the release branch instead of `main`.

## Description
Rancher added a feature for users to customize webhook values in https://github.com/rancher/rancher/pull/42060, But this feature was never documented. So users are unable to use it effectively.

### CODE REVIEW

File: docs/reference-guides/rancher-webhook.md [hunk 1]
```diff
@@ -48,20 +48,56 @@ To bypass the webhook, impersonate both the `rancher-webhook-sudo` service accou
 kubectl create -f example.yaml --as=system:serviceaccount:cattle-system:rancher-webhook-sudo --as-group=system:masters
 ```
 
+## Customizing Rancher-Webhook Configuration
+You can add custom Helm values when you install Rancher-Webhook via Helm. During a Helm install of the Rancher-Webhook chart, Rancher checks for custom Helm values. These custom values must be defined in a ConfigMap named `rancher-config`, in the `cattle-system` namespace, under the data key, `rancher-webhook`. The value of this key must be valid YAML.
```
Comment (martyav) [inline docs/reference-guides/rancher-webhook.md:52]: See https://github.com/rancher/rancher-docs/pull/1081/files#r1463621623 for edit suggestion history

File: docs/reference-guides/rancher-webhook.md [hunk 2]
```diff
@@ -48,20 +48,56 @@ To bypass the webhook, impersonate both the `rancher-webhook-sudo` service accou
...
+  labels:
+    app.kubernetes.io/part-of: "rancher"
+data:
+  rancher-webhook: '{"port": 9553, "priorityClassName": "system-node-critical"}'
+
+```
+Rancher redeploys the Rancher-Webhook chart when changes to the ConfigMap values are detected.
+
+### Customizing Rancher-Webhook During Rancher Installation
+When you use Helm to install the Rancher chart, you can add custom Helm values to the Rancher-Webhook of the local cluster. All values in the Rancher-Webhook chart are accessible as nested variables under the `webhook` name.
```
Comment (martyav) [inline docs/reference-guides/rancher-webhook.md:68]: https://github.com/rancher/rancher-docs/pull/1081/files#r1463652112

File: docs/reference-guides/rancher-webhook.md [hunk 3]
```diff
@@ -48,20 +48,56 @@ To bypass the webhook, impersonate both the `rancher-webhook-sudo` service accou
 kubectl create -f example.yaml --as=system:serviceaccount:cattle-system:rancher-webhook-sudo --as-group=system:masters
 ```
 
+## Customizing Rancher-Webhook Configuration
```
Comment (martyav) [inline docs/reference-guides/rancher-webhook.md:51]:
```suggestion
## Customizing Rancher-Webhook Configuration

```

File: docs/reference-guides/rancher-webhook.md [hunk 4]
```diff
@@ -48,20 +48,56 @@ To bypass the webhook, impersonate both the `rancher-webhook-sudo` service accou
...
+kind: ConfigMap
+metadata:
+  name: rancher-config
+  namespace: cattle-system
+  labels:
+    app.kubernetes.io/part-of: "rancher"
+data:
+  rancher-webhook: '{"port": 9553, "priorityClassName": "system-node-critical"}'
+
+```
```
Comment (martyav) [inline docs/reference-guides/rancher-webhook.md:64]: Suggest adding an extra line break after the triple backticks. I'd add it myself but Github review comments tend to eat the triple backtick syntax...

File: docs/reference-guides/rancher-webhook.md [hunk 5]
```diff
@@ -48,20 +48,56 @@ To bypass the webhook, impersonate both the `rancher-webhook-sudo` service accou
...
+Rancher redeploys the Rancher-Webhook chart when changes to the ConfigMap values are detected.
+
+### Customizing Rancher-Webhook During Rancher Installation
```
Comment (martyav) [inline docs/reference-guides/rancher-webhook.md:67]:
```suggestion
### Customizing Rancher-Webhook During Rancher Installation

```

**Review by martyav — CHANGES_REQUESTED** [review]
Some small spacing issues, plus links to the original edit suggestions in the old PR

**Review by martyav — APPROVED** [review]
Approved, added versioning

**Comment by martyav** [comment]
@JonCrowther should we merge or do we need a technical review?

**Comment by JonCrowther** [comment]
@martyav would like one technical review. @nflynt has been added

**Review by nflynt — APPROVED** [review]
:+1:

**Pull request merged**